tests/test.linter.js

```diff
@@ -883,37 +883,28 @@ describe('Linter.extractMetadata()', function() {
       });
   });
 
-  // Uses our angular-unadvised-library XPI, with the following file layout:
-  //
-  // - bootstrap.js
-  // - data/
-  //   - angular-1.5.3.js (Angular)
-  //   - change-text.js
-  //   - empty.js (empty file)
-  //   - jquery-2.1.4.min.js (minified jQuery)
-  // - index.js
-  // - install.rdf
-  // - package.json
-  // - README.md
   it('should flag unadvised JS libraries in an XPI.', () => {
     var addonLinter = new Linter({
-      _: ['tests/fixtures/angular-unadvised-library.xpi'],
+      _: ['fake.xpi'],
     });
-    var markUnadvisedSpy = sinon.spy(addonLinter, '_markBannedLibs');
+    var unadvised_libs = ['test_unadvised_fake_lib.js'];
+    var metadata = {
+      jsLibs : {
+        'data/unadvised_fake_lib.js': 'test_unadvised_fake_lib.js',
+        'data/jquery-2.1.4.min.js': 'jquery.2.1.4.jquery.min.js',
+      },
+    };
 
-    return addonLinter.extractMetadata({_console: fakeConsole})
-      .then((metadata) => {
-        assert.ok(markUnadvisedSpy.called);
-        assert.equal(Object.keys(metadata.jsLibs).length, 2);
-        assert.deepEqual(metadata.jsLibs, {
-          'data/angular-1.5.3.js': 'angularjs.1.5.3.angular.js',
-          'data/jquery-2.1.4.min.js': 'jquery.2.1.4.jquery.min.js',
-        });
+    var result = addonLinter._markBannedLibs(metadata, unadvised_libs);
+    assert.equal(Object.keys(result.jsLibs).length, 2);
+    assert.deepEqual(result.jsLibs, {
```

Comment:
You set this above, maybe you can just point to metadata.jsLibs instead of duplicating it?